src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -142,6 +142,56 @@ private static TimeZoneInfo GetLocalTimeZoneCore()
             return Utc;
         }
 
+        private static TimeSpan GetCacheLocalUtcOffset(DateTime dateTime, TimeZoneInfoOptions flags)
+        {
+                CachedData cachedData = s_cachedData;
+                return cachedData.Local.GetUtcOffset(dateTime, flags, cachedData);
+        }
+
+        private static TimeSpan? _localUtcOffset;
+        private static object _localUtcOffsetLock = new();
+        private static Thread? _loadAndroidTZData;
+        // Shortcut for TimeZoneInfo.Local.GetUtcOffset
+        // On Android, loading AndroidTZData while obtaining cachedData.Local is expensive for startup.
+        // We introduce a fast result for GetLocalUtcOffset that relies on the date time offset being
+        // passed into monovm_initialize(_preparsed) from Java in seconds as LOCAL_DATE_TIME_OFFSET.
+        // However, to handle timezone changes during the app lifetime, AndroidTZData needs to be loaded.
+        // The fast path is initially used, and we start a background thread to get cachedData.Local
+        internal static TimeSpan GetLocalUtcOffset(DateTime dateTime, TimeZoneInfoOptions flags)
+        {
+            if (_localUtcOffset != null) // The background thread finished, the cache is loaded.
+                return GetCacheLocalUtcOffset(dateTime, flags);
+
+            if (_localUtcOffset == null && _loadAndroidTZData == null) // The cache isn't loaded and no background thread has been created
+            {
+                lock (_localUtcOffsetLock)
+                {
+                    // GetLocalUtcOffset may be called multiple times before a cache is loaded and a background thread is running,
+                    // once the lock is available, check for a cache and background thread.
+                    if (_localUtcOffset != null)
+                        return GetCacheLocalUtcOffset(dateTime, flags);
+
+                    if (_loadAndroidTZData == null)
+                    {
+                        _loadAndroidTZData = new Thread(() => {
+                            Thread.Sleep(1000);
+                            _localUtcOffset = GetCacheLocalUtcOffset(dateTime, flags);
+                        });
+                        _loadAndroidTZData.IsBackground = true;
+                        _loadAndroidTZData.Start();
+                    }
+                }
+            }
+
+            object? localDateTimeOffset = AppContext.GetData("LOCAL_DATE_TIME_OFFSET");
```

Comment:
Is there some prefix that we need to use? 
From the docs it looks like normally the names are namespaced like classes https://docs.microsoft.com/en-us/dotnet/core/runtime-config/

So maybe `System.TimeZoneInfo.LocalDataTimeOffset` ?

Reply:
Thats a good point, as in the App Context Property's Key should follow that format for usage in the managed side? I think you're right, I was following convention that the other properties use, but then again, it doesn't seem like any of the other properties are directly used in the managed side. It will stand out here https://github.com/xamarin/xamarin-android/blob/a689fb9b0e1459c95115b4c9949cb7ec4068f2be/src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.RuntimeConfig.targets#L23-L30, but the new naming will make it more clear where the property is being used right?

Reply:
@vitek-karas Do we keep track of important AppContext keys somewhere? Is there a convention for naming them?  Some way to avoid clashing with third-party names?

Reply:
There's no convention which we all agreed on. For feature switches we use the "looks like managed namespace qualified name" - and if there's a managed counterpart (property) we typically use the same name. https://github.com/dotnet/runtime/blob/main/docs/workflow/trimming/feature-switches.md

Other knobs also tend to follow the "looks like managed" convention, but not all.

I think we should use that convention everywhere we can - just like namespaces it leads to pretty much guaranteed uniqueness. But that's just my opinion.

Reply:
Named it to `System.TimeZoneInfo.LocalDateTimeOffset`